tests/test_runtime.py

```diff
@@ -0,0 +1,736 @@
+        program_id = upload_vqe_runner(hub="ibm-q-startup", group="xanadu", project="reservations")
+
+        job = vqe_runner(
+            program_id=program_id,
+            backend="ibmq_qasm_simulator",
+            hamiltonian=hamiltonian,
+            ansatz="EfficientSU2",
+            x0=[3.97507603, 3.00854038],
+            shots=shots,
+            kwargs={"hub": "ibm-q-startup", "group": "ibm-q-startup", "project": "reservations"},
+        )
+
+        provider = IBMQ.get_provider(hub="ibm-q-startup", group="xanadu", project="reservations")
+        delete_vqe_runner(provider=provider, program_id=program_id)
+
+        assert isinstance(job.intermediate_results, dict)
+        assert "nfev" in job.intermediate_results
+        assert "parameters" in job.intermediate_results
+        assert "function" in job.intermediate_results
+        assert "step" in job.intermediate_results
+        assert "accepted" in job.intermediate_results
+
+    @pytest.mark.parametrize("shots", [8000])
+    def test_qnode(self, token, tol, shots):
+        """Test that we cannot pass a Qnode as ansatz circuit"""
+        IBMQ.enable_account(token)
+
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.qnode(dev)
+        def vqe_circuit(params):
+            qml.RX(params[0], wires=0)
+            qml.RY(params[1], wires=0)
+
+        coeffs = [1, 1]
+        obs = [qml.PauliX(0), qml.PauliZ(0)]
+
+        hamiltonian = qml.Hamiltonian(coeffs, obs)
+        program_id = upload_vqe_runner(hub="ibm-q-startup", group="xanadu", project="reservations")
+
+        with pytest.raises(qml.QuantumFunctionError, match="Must be a callable quantum function."):
+            vqe_runner(
+                program_id=program_id,
+                backend="ibmq_qasm_simulator",
+                hamiltonian=hamiltonian,
+                ansatz=vqe_circuit,
+                x0=[3.97507603, 3.00854038],
+                shots=shots,
+                optimizer="SPSA",
+                optimizer_config={"maxiter": 40},
+                kwargs={
+                    "hub": "ibm-q-startup",
+                    "group": "ibm-q-startup",
+                    "project": "reservations",
+                },
+            )
+
+        provider = IBMQ.get_provider(hub="ibm-q-startup", group="xanadu", project="reservations")
+        delete_vqe_runner(provider=provider, program_id=program_id)
+
+    @pytest.mark.parametrize("shots", [8000])
+    def test_wrong_input(self, token, tol, shots):
+        """Test that we can only give a single vector parameter to the ansatz circuit."""
+        IBMQ.enable_account(token)
+
+        def vqe_circuit(params, wire):
+            qml.RX(params[0], wires=wire)
+            qml.RY(params[1], wires=wire)
+
+        coeffs = [1, 1]
+        obs = [qml.PauliX(0), qml.PauliZ(0)]
+
+        hamiltonian = qml.Hamiltonian(coeffs, obs)
+        program_id = upload_vqe_runner(hub="ibm-q-startup", group="xanadu", project="reservations")
```

Comment:
This seems to be a repeating build-up for tests. It could potentially be organized into a build-up fixture.